### DISCUSSION TITLE
Precision issue with FLOAT DECIMAL64

### DISCUSSION THREAD

**Discussion by ankitsoin** [finos/TimeBase-CE#237] [Q&A]
I have a stream which have columns with datatype FLOAT DECIMAL64.
After loading data to that stream, values gets stored with higher precision level of decimal values (up to 16 decimal places).

Example: 
Value 0.99784 is converted to 0.9978399999999999 
Value 0.6971 is converted to 0.6971000000000001 
while storing in Timebase.

How can we limit to exact original decimal precision values?

**Comment by alex-karpovich** [comment]
please take a look here

https://kb.timebase.info/community/overview/data_types#float

for you I would suggest use DECIMAL(6) or DECIMAL(8), if you are sure that your numbers has fixed precision.

Decimal64 is better for high-precision numbers - guide is here https://github.com/epam/dfp

**Reply by ankitsoin** [reply]
@alex-karpovich 
I can't strictly limit my value to 6 or 8 currently. Even if I take upto DECIMAL(8) then my understanding is Timebase may convert my values like 0.99784 to 0.99783999 ? correct

I saw below in documentation, Data Type Float DECIMAL
**A decimal floating-point number, encoded with automatically chosen precision. This encoding works best with smaller decimal numbers, such as 24.25 or 1234.5556.**

I couldn't understand what it means by encoded with automatically chosen precision, does it means precision will be automatically taken depending upon data values it will have(in terms of decimal places) and then encodes.
Will DECIMAL type work in my use case, as I will be having 1 to 5 or may be upto 6 decimal places price values data?

**Reply by alex-karpovich** [reply]
Correct, if you do not know precision of your values, use DECIMAL. Precision will be automatically chosen depending on data values.
